### PR TITLE
Fix Nanite Programmer and Nanite Program Hub deconstruction

### PR DESCRIPTION
**Pull request by KathyRyals**
Bug fix for the deconstruction process of the Nanite Programmer and Nanite Program Hub

## About The Pull Request

For some reasons , the machines could not be deconstructed via the standard screwdriver > crowbar combo. This PR fixes it.

## Why It's Good For The Game

Fixing bugs is always good.

## Changelog
:cl: Kathy Ryals
fix: The Nanite Programmer and Nanite Program Hub can now be properly deconstructed.
/:cl:

### CODE REVIEW

File: code/modules/research/nanites/nanite_program_hub.dm [hunk 1]
```diff
@@ -37,6 +37,10 @@
 			if(disk)
 				eject(user)
 			disk = N
+	else if (default_deconstruction_screwdriver(user, "nanite_program_hub", "nanite_program_hub", I))
```
Comment (Qustinnus) [inline code/modules/research/nanites/nanite_program_hub.dm:40]: put this in screwdriver_act()

File: code/modules/research/nanites/nanite_program_hub.dm [hunk 2]
```diff
@@ -37,6 +37,10 @@
...
+		return
+	else if(default_deconstruction_crowbar(I))
```
Comment (Qustinnus) [inline code/modules/research/nanites/nanite_program_hub.dm:42]: put this in crowbar_act

File: code/modules/research/nanites/nanite_programmer.dm [hunk 3]
```diff
@@ -23,6 +23,10 @@
 				eject(user)
 			disk = N
 			program = N.program
+	else if (default_deconstruction_screwdriver(user, "nanite_programmer", "nanite_programmer", I))
```
Comment (Qustinnus) [inline code/modules/research/nanites/nanite_programmer.dm:26]: ditto

File: code/modules/research/nanites/nanite_programmer.dm [hunk 4]
```diff
@@ -23,6 +23,10 @@
...
+		return
+	else if(default_deconstruction_crowbar(I))
+		return
 	else
 		..()
```
Comment (Qustinnus) [inline code/modules/research/nanites/nanite_programmer.dm:31]: ditto

Comment (Qustinnus) [reply code/modules/research/nanites/nanite_programmer.dm:31]: clicked the wrong line like a dumbass ignore this one

File: code/modules/research/nanites/nanite_programmer.dm [hunk 5]
```diff
@@ -23,6 +23,10 @@
 				eject(user)
 			disk = N
 			program = N.program
+	else if (default_deconstruction_screwdriver(user, "nanite_programmer", "nanite_programmer", I))
+		return
+	else if(default_deconstruction_crowbar(I))
```
Comment (Qustinnus) [inline code/modules/research/nanites/nanite_programmer.dm:28]: ditto

**Comment by Timberpoes** [comment]
Again, another collaboration with Kathy.

I've implemented the screwdriver_act and crowbar_act overrides.

Since these machines were missing icon_states for having their circuits exposed with a screwdriver, I have done some rough extra ones.

Here's gifs of my testing setup:
8dHUVoEedI
i5bxi0sxtW
mBKTgnoSLQ

File: code/modules/research/nanites/nanite_program_hub.dm (hunk 1 above)
Comment (Timberpoes) [reply code/modules/research/nanites/nanite_program_hub.dm]: Done.

File: code/modules/research/nanites/nanite_program_hub.dm (hunk 2 above)
Comment (Timberpoes) [reply code/modules/research/nanites/nanite_program_hub.dm]: Done.

File: code/modules/research/nanites/nanite_programmer.dm (hunk 3 above)
Comment (Timberpoes) [reply code/modules/research/nanites/nanite_programmer.dm]: Done.

File: code/modules/research/nanites/nanite_programmer.dm (hunk 5 above)
Comment (Timberpoes) [reply code/modules/research/nanites/nanite_programmer.dm]: Done.

**Comment by Rohesie** [comment]
Good PR, thanks for the fix!

**Pull request merged**